langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal

```diff
@@ -704,7 +706,60 @@ function testShiftOnTupleWithoutValuesForRestParameter() {
     var message = err.detail()["message"];
     string detailMessage = message is error ? message.toString() : message.toString();
     assertValueEquality("{ballerina/lang.array}OperationNotSupported", err.message());
-    assertValueEquality("shift() not supported on type '[int,int...]'", detailMessage);
+    assertValueEquality("the number of members in a tuple value should be greater " +
+    "than the member types of the tuple to perform a 'shift' operation", detailMessage);
+}
+
+function testShiftOnTupleWithVariableInherentTypes() {
+    [int, int, string...] tupleWithVariableInherentTypes1 = [0, 1, "hello"];
+    [anydata, anydata, anydata...] tempTuple1 = tupleWithVariableInherentTypes1;
+
+    [int, string, string...] tupleWithVariableInherentTypes2 = [0, "hello", "world"];
+    [anydata, anydata, anydata...] tempTuple2 = tupleWithVariableInherentTypes2;
+
+    var fn1 = function() {
+        var x = tempTuple1.shift();
+    };
+
+    var fn2 = function() {
+        var y = tempTuple2.shift();
+    };
+
+    error? res1 = trap fn1();
+    error? res2 = trap fn2();
+    assertTrue(res1 is error);
+    assertTrue(res2 is error);
+
+    error err1 = <error>res1;
+    error err2 = <error>res2;
+    var message1 = err1.detail()["message"];
+    var message2 = err1.detail()["message"];
+    string detailMessage1 = message1 is error ? message1.toString() : message1.toString();
+    string detailMessage2 = message2 is error ? message2.toString() : message2.toString();
+    assertValueEquality("{ballerina/lang.array}InherentTypeViolation", err1.message());
+    assertValueEquality("{ballerina/lang.array}InherentTypeViolation", err2.message());
+    assertValueEquality("incompatible types: expected 'int', found 'string'", detailMessage1);
+    assertValueEquality("incompatible types: expected 'int', found 'string'", detailMessage2);
+}
+
+function testShiftOnTupleWithProperConditions() {
+    [int, int, int...] properTuple = [0, 1, 3];
```

Comment:
Lets add one test case with some structured type like a record

Reply:
Thank you for the suggestion. Added with 764529a65fc3e0fe3b485eeae6004573d91e29e4